src/main/java/com/openmrs/migrator/core/services/impl/PDIServiceImpl.java

```diff
@@ -21,53 +20,57 @@
 
   private static Logger LOG = LoggerFactory.getLogger(PDIServiceImpl.class);
 
-  private final String[] transformations = {"structure/merge-patient.ktr"};
-
-  @Autowired private KettleUtils kettleUtils;
-
   @Autowired private FileIOUtilities fileIOUtilities;
 
   @Override
-  public void runTransformation(InputStream xmlStream) throws KettleException {
-
-    kettleUtils.loadProperties();
+  public boolean runJob(InputStream xmlStream) throws KettleException {
 
     KettleEnvironment.init();
 
-    TransMeta transMeta = new TransMeta(xmlStream, null, true, null, null);
+    JobMeta jobMeta = new JobMeta(xmlStream, null, null);
+
+    Job job = new Job(null, jobMeta);
+    job.setLogLevel(LogLevel.BASIC);
 
-    Trans transformation = new Trans(transMeta);
-    transformation.setLogLevel(LogLevel.BASIC);
+    String name = job.getName();
+    LOG.info("Starting Job " + name);
 
-    String name = transformation.getName();
-    LOG.info("Starting transformation " + name);
+    Result result = job.execute(1, null);
 
-    transformation.execute(new String[0]);
-    transformation.waitUntilFinished();
+    job.waitUntilFinished();
 
-    Result result = transformation.getResult();
+    boolean executedSucessfully = false;
 
     String outcome =
         String.format(
-            "Transformation %s executed %s",
+            "Job %s executed %s",
             name,
             (result.getNrErrors() == 0
                 ? "successfully"
                 : "with " + result.getNrErrors() + " errors"));
 
+    if (result.getNrErrors() == 0) {
+      outcome = String.format("Job %s executed %s", name, "successfully");
+
+      executedSucessfully = true;
+      LOG.info(outcome);
+      return executedSucessfully;
+    }
+
+    outcome = String.format("Job %s executed %s", name, "with " + result.getNrErrors() + " errors");
     LOG.info(outcome);
+
+    return executedSucessfully;
   }
 
   @Override
-  public void mergeOpenMRS() {
+  public void mergeOpenMRS(String[] jobs) throws IOException {
```

Comment:
We are not merging anymore (at least not yet) so this is not technically an accurate name for the function. 

Also I think we can move this to the MigratorApplication class since all it is doing is loading up a job file and running the `runJob` function. Lets let the application 'run' the app instead of moving this orchestration type logic into the PDI service.

What do you think?

Reply:
Sounds good!
So we are renaming the method "_mergeOpenMRS_"  to "_runAllJobs_" for example,  and move the logic to **MigratorApplication.class**

Reply:
Done